File: NewsDaily/newsloader/src/main/java/com/andruid/magic/newsloader/server/RetrofitService.java
package com.andruid.magic.newsloader.server;

import com.andruid.magic.newsloader.model.ApiResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface RetrofitService {
    @GET("/headlines")
    Call<ApiResponse> getHeadlines(@Query("country") String country, @Query("page") int page,
                                   @Query("page_size") int pageSize);
}

File: NewsDailyJava/newsloader/src/main/java/com/andruid/magic/newsloader/server/RetrofitService.kt
package com.andruid.magic.newsloader.server

import com.andruid.magic.newsloader.model.ApiResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface RetrofitService {
    @GET("/headlines")
    fun getHeadlines(@Query("country") country: String, @Query("category") category: String,
                     @Query("page") page: Int, @Query("page_size") pageSize: Int
    ): Call<ApiResponse>

    @GET("/articles")
    fun getArticles(@Query("language") language: String, @Query("query") query: String,
                    @Query("page") page: Int, @Query("page_size") pageSize: Int
    ): Call<ApiResponse>
}